pkg/services/mysqldb/provision.go

```diff
@@ -30,6 +32,39 @@ func (s *serviceManager) ValidateProvisioningParameters(
 			fmt.Sprintf(`invalid sslEnforcement option: "%s"`, pp.SSLEnforcement),
 		)
 	}
+
+	startIP := net.ParseIP(pp.FirewallIPStart)
+	if pp.FirewallIPStart != "" && net.ParseIP(pp.FirewallIPStart) == nil {
+		return service.NewValidationError(
+			"firewallStartIpAddress",
+			fmt.Sprintf(`invalid firewallStartIpAddress option: "%s"`, pp.SSLEnforcement),
+		)
+	}
+
+	endIP := net.ParseIP(pp.FirewallIPEnd)
+	if pp.FirewallIPEnd != "" && net.ParseIP(pp.FirewallIPEnd) == nil {
+		return service.NewValidationError(
+			"firewallEndIpAddress",
+			fmt.Sprintf(`invalid firewallEndIpAddress option: "%s"`, pp.SSLEnforcement),
+		)
+	}
+
+	if startIP != nil && endIP == nil {
```

Comment:
This seems to catch the scenario where a start ip was provided, but no end ip was provided. But the opposite scenario is equally a possibility and looks unaccounted for.

Maybe we can actually move the check for this higher up in this validation function (before we do any parsing) and just have it be that if one is defined, they _both_ must be?